### PR TITLE
chore: merge infallible errors, improve expect statements and use {:#?} for anyhow error formatting

### PR DESCRIPTION
**Pull request by rumenov**

### CODE REVIEW

File: rs/p2p/quic_transport/src/connection_manager.rs [hunk 1]
```diff
@@ -261,6 +258,25 @@ impl ConnectionManager {
         self.node_id < *dst
     }
 
+    // Conditions under which the node can start outbound connecting attempt
+    // - the node is a designated dialer
+    // - peer is in the subnet
+    // - this node is not part of the subnet (can happen when a node is removed from the subnet)
+    // - there is no connect attemped
+    // - there is no established connections
```
Comment (DSharifi) [inline rs/p2p/quic_transport/src/connection_manager.rs:266]: Nit:
Use triple backlash for docs

File: rs/p2p/quic_transport/src/connection_manager.rs [hunk 2]
```diff
@@ -261,6 +258,25 @@ impl ConnectionManager {
         self.node_id < *dst
     }
 
+    // Conditions under which the node can start outbound connecting attempt
+    // - the node is a designated dialer
+    // - peer is in the subnet
+    // - this node is not part of the subnet (can happen when a node is removed from the subnet)
+    // - there is no connect attemped
```
Comment (DSharifi) [inline rs/p2p/quic_transport/src/connection_manager.rs:265]: ?
```suggestion
    // - there is no connection attemped
```

File: rs/p2p/quic_transport/src/connection_manager.rs (hunk 1 above)
Comment (DSharifi) [inline rs/p2p/quic_transport/src/connection_manager.rs:266]:
```suggestion
    // - there is no established connection
```

File: rs/p2p/quic_transport/src/connection_manager.rs [hunk 3]
```diff
@@ -261,6 +258,25 @@ impl ConnectionManager {
         self.node_id < *dst
     }
 
+    // Conditions under which the node can start outbound connecting attempt
+    // - the node is a designated dialer
+    // - peer is in the subnet
+    // - this node is not part of the subnet (can happen when a node is removed from the subnet)
```
Comment (DSharifi) [inline rs/p2p/quic_transport/src/connection_manager.rs:264]: The node should be in the subnet. This is also the condition you check in the code.

File: rs/p2p/quic_transport/src/connection_manager.rs [hunk 4]
```diff
@@ -366,7 +382,7 @@ impl ConnectionManager {
             .server_config(subnet_nodes, self.topology.latest_registry_version())
         {
             Ok(rustls_server_config) => {
-                let quic_server_config = QuicServerConfig::try_from(rustls_server_config).unwrap();
+                let quic_server_config = QuicServerConfig::try_from(rustls_server_config).expect("Conversion from RustTls config to Quinn config must succeed as long as this library and quinn use the same RustTls versions.");
```
Comment (DSharifi) [inline rs/p2p/quic_transport/src/connection_manager.rs:385]: Thank you for this

**Review by DSharifi — APPROVED** [review]

File: rs/p2p/quic_transport/src/connection_manager.rs (hunk 3 above)
Comment (rumenov) [reply rs/p2p/quic_transport/src/connection_manager.rs:264]: It is checked, right ?

Comment (DSharifi) [reply rs/p2p/quic_transport/src/connection_manager.rs]: But your comment bullet point says the opposite?

**Pull request merged**